android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -653,14 +651,9 @@ public void receiveCommand(@NonNull WebView root, String commandId, @Nullable Re
         root.stopLoading();
         break;
       case "postMessage":
-        try {
-          RNCWebView reactWebView = (RNCWebView) root;
-          JSONObject eventInitDict = new JSONObject();
-          eventInitDict.put("data", args.getString(0));
-          reactWebView.evaluateJavascriptWithFallback("document.dispatchEvent(new MessageEvent('message', " + eventInitDict.toString() + "));");
-        } catch (JSONException e) {
-          throw new RuntimeException(e);
-        }
+        RNCWebView reactWebView_ = (RNCWebView) root;
+        String encodedString = Base64.encodeToString(args.getString(0).getBytes(), Base64.DEFAULT | Base64.NO_WRAP);
+        reactWebView_.evaluateJavascriptWithFallback("document.dispatchEvent(new MessageEvent('message', {data: Buffer.from('" + encodedString + "', 'base64')}));");
```

Comment:
Note: the original idea was to call `Buffer.from(string, 'base64').toString()` to decode the base64 to a string, but I forgot to add the `toString` call. I'll fix this, but the code worked anyway.

> Why is it safe to use global Buffer inside webview?

That's a good point. It worked when I tested it, even though it doesn't make much sense to use `Buffer` in the browser context. I did some experiments, and suspect we can use `Buffer` because we're using a polyfill in the injected provider. @ChALkeR do you think that defining `window.Buffer` from our provider could be dangerous somehow? If that's the case, we should fix that in a separate PR.

Reply:
`atob()` without using `TextDecoder` breaks on non-ascii symbols.

Reply:
![image](https://user-images.githubusercontent.com/291301/226716429-6a7394b8-90cd-49da-a7d3-007e5620a7f8.png)

Reply:
Yes, let's think about some alternative approach that also won't hit performance as `TextDecoder` would.